src/main/generic/SDL_main_callbacks.c

```diff
@@ -0,0 +1,62 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2023 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+
+#include "SDL_internal.h"
+#include "../../video/SDL_sysvideo.h"
+
+int SDL_EnterAppMainCallback(int argc, char* argv[], SDL_AppInit_func appinit, SDL_AppIterate_func appiter, SDL_AppQuit_func appquit)
+{
+    int rc = appinit(argc, argv);
+
+    const Uint64 ns60fps = 1000000000 / 60;
+    Uint64 next_iteration = SDL_GetTicksNS() + ns60fps;
+    while (rc == 0) {
+        // !!! FIXME: this can be made more complicated if we decide to
+        // !!! FIXME: optionally hand off callback responsibility to the
+        // !!! FIXME: video subsystem (for example, if Wayland has a
+        // !!! FIXME: protocol to drive an animation loop, maybe we hand
+        // !!! FIXME: off to them here if/when the video subsystem becomes
+        // !!! FIXME: initialized).
+        rc = appiter();
+
+        // if there's no window, try to run at about 60fps. This makes this
+        //  not eat all the CPU in simple things like loopwave. If there's
```

Comment:
Shouldn't this be the responsibility of the application? Say I want to use SDL on a headless display, having a 60fps limit would be a bit annoying if I want my application to finish as fast as possible.

Reply:
It's an optional feature. If you want to control the mainloop and have deep concerns about framerate, you don't have to use it.

Reply:
Oh, wait, you meant the no-window limiter specifically.

I can add a hint in case this scenario happens.

Reply:
> Oh, wait, you meant the no-window limiter specifically.

Yeah. More specifically the scenario would be an integration test on the game with the `dummy` video driver, which IIRC can't be used to create a (dummy) window.

Reply:
I think the dummy driver should be able to make a fake window (but maybe there's a bug?), but it makes sense to add a hint in case you want to run at a higher/lower framerate, or just run as fast as possible, so I'll fix that.

Reply:
Resolved by 535d42befbcd946bf6f5226aeb82296485c6630d